src/octopeer-github/content/ElementSelectionBehaviour/ElementSelectionBehaviourFactory.ts

```diff
@@ -0,0 +1,57 @@
+/// <reference path="GenericElementSelectionBehaviour.ts"/>
+/**
+ * Created by Mathias on 2016-05-31.
+ */
+class ElementSelectionBehaviourFactory {
+
+    /**
+     * The ElementSelectionBehaviourData list, sorted by ElementID.
+     * Sorting is done in the constructor.
+     */
+    private elementSelectionBehaviourData: ElementSelectionBehaviourData[] = [];
+
+    /**
+     * Create a ElementSelectionBehaviourFactory and puts the ESBData objects at the index
+     * corresponding to their ElementID (for easier retrieval of ESBData).
+     */
+    constructor() {
+        let data: ElementSelectionBehaviourData;
+
+        for (data of unsortedElementSelectionBehaviourData) {
+            this.elementSelectionBehaviourData[data.elementID.getElementID()] = data;
+        }
+    }
+
+    /**
+     * Creates a GenericElementSelectionBehaviour by finding the right
+     * ElementSelectionBehaviourData and passing this and the database to
+     * the GenericElementSelectionBehaviour constructor.
+     * @param database The database to be used
+     * @param ID The ElementID the data should match
+     * @returns {any}
+     */
+    public create(database: DatabaseAdaptable, ID: ElementID) {
+        let elementSelectionBehaviourData = this.findElementSelectionBehaviourData(ID);
+        if (elementSelectionBehaviourData != null) {
+            return new GenericElementSelectionBehaviour(database, elementSelectionBehaviourData);
+        }
+        return null;
+    }
+
+    /**
+     * Get the sorted ElementSelectionBehaviourData list
+     * @returns {ElementSelectionBehaviourData[]}
+     */
+    public getElementSelectionBehaviourData() {
+        return this.elementSelectionBehaviourData;
+    }
```

Comment:
This method is not used anywhere. Please consider removing it, since other classes should have enough with the "find" method.